app/ide-desktop/lib/dashboard/src/components/AriaComponents/Button/Button.tsx

```diff
@@ -15,73 +15,204 @@ import SvgMask from '#/components/SvgMask'
+  delete: 'bg-delete text-white',
+  icon: 'opacity-50 hover:opacity-100',
+  submit: 'bg-invite text-white opacity-80 hover:opacity-100',
+  outline: 'border-primary text-primary font-bold hover:border-primary-90',
+}
+
+const CLASSES_FOR_ROUNDING: Record<Roundings, string> = {
+  full: 'rounded-full',
+  large: 'rounded-lg',
+  medium: 'rounded-md',
+  none: 'rounded-none',
+  small: 'rounded-sm',
+  xlarge: 'rounded-xl',
+}
+
+const ICON_POSITION: Record<IconPosition, string> = {
+  start: '',
+  end: 'flex-row-reverse',
 }
 
 /** A button allows a user to perform an action, with mouse, touch, and keyboard interactions. */
-export function Button(props: ButtonProps) {
-  const { className, children, variant, icon, loading = false, ...ariaButtonProps } = props
+export const Button = React.forwardRef(function Button(
+  props: ButtonProps,
+  ref: React.ForwardedRef<HTMLButtonElement>
+) {
+  const {
+    className,
+    children,
+    variant,
+    icon,
+    loading = false,
+    isDisabled = loading,
+    iconPosition = 'start',
+    size = 'xsmall',
+    fullWidth = false,
+    rounding = 'large',
+    ...ariaProps
+  } = props
   const focusChildProps = focusHooks.useFocusChild()
 
+  const isLink = ariaProps.href != null
+
+  const Tag = isLink ? aria.Link : aria.Button
+
+  const goodDefaults = isLink ? { rel: 'noopener noreferrer' } : { type: 'button' }
+
   const classes = clsx(
     DEFAULT_CLASSES,
     DISABLED_CLASSES,
     FOCUS_CLASSES,
-    SIZE_CLASSES,
-    CLASSES_FOR_VARIANT[variant]
+    CLASSES_FOR_SIZE[size],
+    CLASSES_FOR_ROUNDING[rounding],
+    CLASSES_FOR_VARIANT[variant],
+    { [LOADING_CLASSES]: loading, [FULL_WIDTH_CLASSES]: fullWidth }
   )
 
   const childrenFactory = (): React.ReactNode => {
-    if (loading) {
-      return <Spinner state={spinnerModule.SpinnerState.loadingMedium} size={16} />
-    } else if (variant === 'icon' && icon != null) {
+    // Icon only button
+    if (variant === 'icon' && icon != null) {
       return (
-        <div className={EXTRA_CLICK_ZONE_CLASSES}>
-          <SvgMask src={icon} />
-        </div>
+        <aria.Text className={EXTRA_CLICK_ZONE_CLASSES}>
+          <SvgMask src={icon} className="flex-none" />
+        </aria.Text>
       )
     } else {
-      return <>{children}</>
+      // Default button
+      return (
+        <aria.Text className={clsx('flex items-center gap-2', ICON_POSITION[iconPosition])}>
+          {icon != null && <SvgMask src={icon} className="flex-none" />}
+          <>{children}</>
+        </aria.Text>
+      )
     }
   }
 
   return (
-    <aria.Button
-      {...aria.mergeProps<aria.ButtonProps>()(ariaButtonProps, focusChildProps, {
-        className: values =>
-          tailwindMerge.twMerge(
-            classes,
-            typeof className === 'function' ? className(values) : className
-          ),
+    <Tag
+      // @ts-expect-error eventhough typescript is complaining about the type of ariaProps, it is actually correct
```

Comment:
@somebody1234 do you have any ideas on how can we satisfy the typescript here?

Reply:
i guess the conventional solution is to just extract the props and children to variables, and then to explicitly create an `<aria.Link` or  `<aria.Button` in a `switch` or conditional (or ternary)